lib/ansible/modules/crypto/openssl_csr.py

```diff
@@ -431,7 +431,7 @@ def _check_subject(csr):
 
         def _check_subjectAltName(extensions):
             altnames_ext = next((ext for ext in extensions if ext.get_short_name() == b'subjectAltName'), '')
-            altnames = [altname.strip() for altname in str(altnames_ext).split(',')]
+            altnames = [altname.strip() for altname in str(altnames_ext).split(',') if altname.strip() if altname.strip()]
```

Comment:
It isn't a bug (so I'm going to merge this), but I do notice a few things about this list comprehension that I think could be changed.  `if altname.strip()` is repeated twice at the end.  I think that's probably a typo as it is equivalent to `if altname.strip() and altname.strip()` which is redundant checking.

If you don't want to duplicate calling strip() on altname, you can also do that as part of setting altname.  For readability, I'd split it up into two lines like this:
``` python
             altnames = (altname.strip() in str(altnames_ext).split(','))
             altnames = [altname for altname in altnames if altname]
```

(Which seems to fit in with the style of this function as the next line further modifies altnames in a third list comprehension/generator expression).

Reply:
Actually, combining it with the next line, I'd do it like this:
``` python
             altnames = (altname.strip() in str(altnames_ext).split(','))
             altnames = [name if not name.startswith('IP Address:') else "IP:" + name.split(':', 1)[1] for name in altnames if altname]
```

Reply:
Good catch! The double `if` was not intentional, I must have completely missed that. I had to do the cherry-picking mostly manual, since the module changed too much in `devel`.